Update viewport scaling demo to 4.0.

rcorre opened the pull request:
Fixes #856.

The 3d_scaling demo was updated to use the built-in 3D scaling mechanisms
in the Forward+ renderer.

A new 3d_scaling_compat demo was added to demonstrate the legacy
SubViewport-based scaling method for the Compatibility renderer.

<!--
Only submit a pull request if all of the following conditions are met:

* It must work with the latest Godot version of the branch you're submitting to.

* It must follow all of the Godot style guides, including the GDScript
  style guide and the C# style guide.

* The demo should not be overcomplicated. Simplicity is usually preferred.

* If you are submitting a new demo, please ensure that it includes a
  README file similar to the other demos.

* If you are submitting a copy of a demo translated to C# etc:

    * Please ensure that there is a good reason to have this demo translated.
      We don't want to have multiple copies of every single project.

    * Please ensure that the code mirrors the original closely.

    * In the project.godot file and in the README, include "with C#" etc in
      the title, and also include a link to the original in the README.
-->

rcorre commented on viewport/3d_scaling_compat/README.md:
```diff
@@ -0,0 +1,21 @@
+
+Renderer: Compatibility
+
+Check out this demo on the asset library: https://godotengine.org/asset-library/asset/586
```
Do we need a new asset library entry?

Calinou replied:
Yes. 4.0 demos haven't been submitted for the asset library yet, but we intend to keep the 3.5 demos available for the foreseeable future.

rcorre commented:
> but the 3D Resolution Scaling (Compatibility) demo needs to resize the SubViewport when the window size changes

Huh, I figured that would happen automatically with **Stretch** enabled. I added this print:

```
get_tree().root.size_changed.connect(func(): prints(get_tree().root.size, viewport_container.size, viewport.size))
```

The viewport is resizing to the parent control, but the way the parent control resizes depends on the `window/stretch/mode` and `window/stretch/aspect` settings in the project. If you set `mode=disabled`, then the viewport container and the viewport auto-adjust to the window size.

leetNightshade commented:
Idk if it's just me, but testing this on Godot 4.1.1.stable.custom_build on Steam Deck doesn't seem to do anything, it draws at the window resolution. I have this same problem in my project that emulates this subviewport project.

AThousandShips commented on viewport/3d_scaling/hud.gd:
```diff
@@ -1,42 +1,25 @@
-# sharper but are slower to render. Values above 1 can be used for supersampling
-# (SSAA), but filtering must be enabled for supersampling to work.
-var scale_factor = 1.0
+# The 3D viewport's shrink factor. For instance, 1 is full resolution,
```
```suggestion

# The 3D viewport's shrink factor. For instance, 1 is full resolution,
```

AThousandShips commented on viewport/3d_scaling/hud.gd:
```diff
@@ -1,42 +1,25 @@
 @onready var scale_label = $VBoxContainer/Scale
 @onready var filter_label = $VBoxContainer/Filter
 
 func _ready():
```
```suggestion

func _ready():
```

AThousandShips commented on viewport/3d_scaling_compat/hud.gd:
```diff
@@ -0,0 +1,26 @@
+extends Control
+
+# The 3D viewport's shrink factor. For instance, 1 is full resolution,
```
```suggestion

# The 3D viewport's shrink factor. For instance, 1 is full resolution,
```

Calinou commented:
https://github.com/godotengine/godot/pull/83976 is landing in 4.3, so I'd prefer we just use the 3D scaling method for all rendering methods now.

rcorre commented:
I updated to only include the builtin scaling demo, and removed the 3D compat demo.
I'm not as familiar with 2D, so I didn't attempt to create a 2D scaling demo.

Calinou commented:
Merged manually with https://github.com/godotengine/godot-demo-projects/commit/722bd1168955b25e498d9b8967caf3970e02877e (which includes some visual and UI improvements). Thanks!

![Screenshot](https://raw.githubusercontent.com/godotengine/godot-demo-projects/master/viewport/3d_scaling/screenshots/3d_scaling.webp)
